Description: Fixed return issues due to memory garbage
 See #429792.  Thanks to A. Costa <agcosta@gis.net> for the useful bug report.
Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=429792
Last-Update: 2007-06-20
Index: num-utils-0.5/numgrep
===================================================================
a/numgrep	2016-08-06 08:56:17.766570099 +0200
b/numgrep	2016-08-06 08:56:17.762570099 +0200
@@ -221,6 +221,7 @@
             $exp = "!($1 % $number)";
         }
     } else {
+        die "Invalid expression: $expression\n";
         $exp = "$number =~ /$expression/";
     }
     print "Exp: '$exp'\n" if ($verbose >= 3);
Index: num-utils-0.5/numprocess
===================================================================
a/numprocess	2016-08-06 08:56:17.766570099 +0200
b/numprocess	2016-08-06 08:56:17.762570099 +0200
@@ -158,6 +158,7 @@
                 $new_number = sqrt($new_number);
             }
         } else {
+            die "Encountered invalid expression $operation in $expression.\n";
             print "Encountered invalid expression $operation in $expression.\n" if ($verbose >= 2);
             return $number;
         }
Index: num-utils-0.5/random
===================================================================
a/random	2016-08-06 08:56:17.766570099 +0200
b/random	2016-08-06 08:56:17.766570099 +0200
@@ -101,6 +101,7 @@
             }
         }
     } else {
+        die "Invalid expression: $expression\n";
         print STDERR "Invalid expression: $expression\n" if ($verbose >= 1);
     }
 }
Index: num-utils-0.5/range
===================================================================
a/range	2016-08-06 08:56:17.766570099 +0200
b/range	2016-08-06 08:56:17.766570099 +0200
@@ -127,6 +127,7 @@
             }
         }
     } else {
+        die "Invalid expression: $expression\n";
         print STDERR "Invalid expression: $expression\n" if ($verbose >= 1);
     }
 }
